src/plots/cartesian/constants.js

```diff
@@ -26,6 +26,7 @@ module.exports = {
     // note that this is more permissive than counterRegex, as
     // id2name, name2id, and cleanId accept "x1" etc
     AX_ID_PATTERN: /^[xyz][0-9]*$/,
+    AX_ID_EXTRAS_PATTERN: /^[xyz]([2-9]|[1-9][0-9]+)?( domain)?$/,
```

Comment:
The way you use this, it ends up replacing x- or y-specific patterns with a generic xyz pattern. See if you can make something more precise like `idRegex.xExtra` - looks like [`counterRegex`](https://github.com/plotly/plotly.js/blob/6b6b475acbc3ff28a0d5603b16583d2444021730/src/lib/regex.js#L21) has a 2nd argument that will allow this, something like `counterRegex('x', '( domain)?')`

Reply:
:+1: